File: revert_hero.js
const fs = require('fs');
let code = fs.readFileSync('frontend/src/App.js', 'utf8');

const heroRegex = /<section[^>]*className="relative pt-32 pb-48[^>]*>[\s\S]*?<\/section>/;

const oldHero = `<section 
              className="relative pt-32 pb-48 overflow-hidden flex items-center justify-center min-h-[80vh] bg-tata-dark bg-cover bg-center"
              style={{ backgroundImage: 'url("https://images.unsplash.com/photo-1577978250009-41fdb4d59a80?auto=format&fit=crop&w=2000&q=80")' }}
            >
              {/* Gradient Overlays for Readability */}
              <div className="absolute inset-0 bg-gradient-to-br from-[#174195]/85 via-[#231F20]/70 to-[#174195]/95 z-0 mix-blend-multiply"></div>
              <div className="absolute inset-0 bg-gradient-to-t from-[#231F20] via-transparent to-transparent z-0 opacity-80"></div>
              
              <div className="absolute inset-0 pointer-events-none overflow-hidden z-0">
                {/* Floating Tea Grades */}
                {FLOATING_GRADES.map((item, idx) => (
                  <div
                    key={idx}
                    className={\`absolute hidden sm:flex px-3 py-1.5 rounded-full bg-white/10 backdrop-blur-sm border border-white/20 shadow-lg text-[9px] font-bold tracking-[0.2em] text-tata-grey-light uppercase animate-float \${item.scale}\`}
                    style={{
                      top: item.top,
                      left: item.left,
                      animationDelay: item.delay,
                    }}
                  >
                    {item.grade}
                  </div>
                ))}
              </div>

              <div className="relative z-10 max-w-5xl mx-auto px-6 w-full flex flex-col items-center text-center gap-10">
                <div className="inline-flex items-center gap-2 px-4 py-2 rounded-full bg-black/40 backdrop-blur-md border border-white/30 shadow-sm transition-all hover:bg-black/60">
                  <Leaf className="w-4 h-4 text-tata-blue-dark" />
                  <span className="text-[10px] sm:text-[11px] uppercase tracking-[0.25em] text-white font-bold">100% Organic · India Organic · USDA · ISO 22000</span>
                </div>
                
                <h1 className="font-serif text-[48px] sm:text-[64px] lg:text-[76px] font-medium tracking-tight text-white leading-[1.05] max-w-4xl mx-auto drop-shadow-2xl">
                  Single-Estate <span className="italic text-tata-blue-dark">Organic Tea</span> from the Foothills of Arunachal
                </h1>
                
                <div className="w-20 h-1 bg-tata-blue-dark rounded-full shadow-[0_0_10px_rgba(212,175,55,0.5)]" />
                
                <p className="text-lg sm:text-xl text-tata-grey-light font-normal leading-relaxed max-w-2xl mx-auto drop-shadow-md">
                  Cultivated, plucked, and processed on our own <strong className="font-medium text-white">1,200-bigha</strong> organic garden and <strong className="font-medium text-white">72,000 sq ft</strong> factory in Kharsang. Shipped directly to tea houses and blenders worldwide.
                </p>
                
                <div className="flex flex-col sm:flex-row items-center justify-center gap-5 pt-4">
                  <button
                    onClick={() => setActiveTab("catalog")}
                    data-testid="hero-explore-catalog"
                    className="bg-tata-blue-dark hover:bg-tata-blue-light text-white font-bold px-10 py-5 rounded-full text-xs uppercase tracking-[0.2em] shadow-[0_0_20px_rgba(212,175,55,0.4)] hover:shadow-[0_0_30px_rgba(255,255,255,0.6)] transition-all duration-300 flex items-center gap-3 transform hover:-translate-y-0.5"
                  >
                    <span>Explore Our Tea</span>
                    <ArrowRight className="w-4 h-4 text-tata-dark" />
                  </button>
                  <button
                    onClick={() => setActiveTab("standards")}
                    data-testid="hero-custom-rfq"
                    className="border border-white/40 bg-black/30 hover:bg-white/10 text-white font-bold px-10 py-5 rounded-full text-xs uppercase tracking-[0.2em] shadow-xl hover:shadow-2xl transition-all duration-300 flex items-center gap-3 backdrop-blur-md transform hover:-translate-y-0.5"
                  >
                    <span>Our Operations</span>
                    <ArrowRight className="w-4 h-4 text-white" />
                  </button>
                </div>
              </div>
            </section>`;

code = code.replace(heroRegex, oldHero);
fs.writeFileSync('frontend/src/App.js', code);

File: revert_hero.py
import re

with open('frontend/src/App.js', 'r') as f:
    content = f.read()

old_hero = """
            <section 
              className="relative w-full overflow-hidden flex items-center justify-center bg-black h-screen pt-28 pb-32 sm:py-20"
            >
              {/* Clean Banner Background Layer */}
              <div className="absolute inset-0 z-0 flex items-center justify-center">
                <img 
                  src="/estate/hero-factory.jpg.png" 
                  alt="Tea Factory Overlooking Garden" 
                  className="w-full h-full object-cover object-center opacity-70 sm:opacity-80 transition-transform duration-700 hover:scale-105" 
                />
                {/* Elegant dark vignette to protect text legibility and blend the contained image edges */}
                <div className="absolute inset-0 bg-gradient-to-b from-[#0a0a0a]/80 via-[#0a0a0a]/50 sm:via-[#0a0a0a]/40 to-[#0a0a0a]/90 pointer-events-none"></div>
                <div className="absolute inset-0 bg-gradient-to-r from-[#0a0a0a]/80 via-transparent to-[#0a0a0a]/80 pointer-events-none opacity-80"></div>
                {/* Extra text protection gradient for mobile */}
                <div className="absolute inset-0 bg-black/40 sm:hidden pointer-events-none"></div>
              </div>
              
              <div className="absolute inset-0 pointer-events-none overflow-hidden z-0">
                {/* Floating Tea Grades */}
                
              </div>

              <div className="relative z-10 max-w-5xl mx-auto px-5 sm:px-8 w-full flex flex-col items-center text-center gap-6 sm:gap-8">
                <div className="inline-flex items-center gap-2 sm:gap-2.5 px-3 py-1.5 sm:px-4 sm:py-2 rounded-none bg-black/50 backdrop-blur-md border border-gray-200 shadow-none transition-all hover:bg-black/70 hover:border-white/40">
                  <Leaf className="w-3.5 h-3.5 sm:w-4 sm:h-4 text-gray-300" />
                  <span className="text-[9px] sm:text-[11px] uppercase tracking-[0.2em] sm:tracking-[0.25em] text-white font-bold">100% Organic · USDA · ISO 22000</span>
                </div>
                
                <h1 className="font-sans font-medium tracking-tight text-4xl sm:text-5xl md:text-6xl text-white leading-tight max-w-4xl mx-auto">
                  Single-Estate <span className="italic text-gray-300">Organic Tea</span> from the Foothills of Arunachal
                </h1>
                
                <div className="w-16 sm:w-24 h-1 sm:h-1.5 bg-transparent rounded-none  opacity-80" />
                
                <p className="text-sm sm:text-base md:text-lg text-gray-400 font-normal leading-relaxed max-w-2xl mx-auto  px-2 sm:px-0">
                  Cultivated, plucked, and processed on our own <strong className="font-medium text-white">1,200-bigha</strong> organic garden and <strong className="font-medium text-white">72,000 sq ft</strong> factory in Kharsang. Shipped directly to tea houses worldwide.
                </p>
                
                <div className="flex flex-col sm:flex-row items-center justify-center gap-3 sm:gap-5 w-full sm:w-auto mt-2">
                  <button
                    onClick={() => setActiveTab("catalog")}
                    data-testid="hero-explore-catalog"
                    className="w-full sm:w-auto bg-black hover:bg-gray-900 text-white font-bold px-6 sm:px-8 py-3.5 sm:py-4 rounded-none text-[11px] sm:text-[12px] uppercase tracking-[0.2em]  hover: transition-all duration-300 flex items-center justify-center gap-2 transform hover:-translate-y-1"
                  >
                    <span>Explore Our Tea</span>
                    <ArrowRight className="w-4 h-4" />
                  </button>
                  <button
                    onClick={() => setActiveTab("standards")}
                    data-testid="hero-custom-rfq"
                    className="w-full sm:w-auto bg-white hover:bg-gray-100 text-black font-bold px-6 sm:px-8 py-3.5 sm:py-4 rounded-none text-[11px] sm:text-[12px] uppercase tracking-[0.2em]  hover: transition-all duration-300 flex items-center justify-center gap-2 backdrop-blur-md transform hover:-translate-y-1"
                  >
                    <span>Our Operations</span>
                    <ArrowRight className="w-4 h-4" />
                  </button>
                </div>
              </div>
            </section>
"""

content = re.sub(
    r'<section\s+className="relative w-full overflow-hidden flex flex-col items-center justify-between bg-black h-screen pb-16 pt-32"\s*>.*?</section>',
    old_hero.strip(),
    content,
    flags=re.DOTALL
)

with open('frontend/src/App.js', 'w') as f:
    f.write(content)
